tools/rosgraph/src/rosgraph/roslogging.py

```diff
@@ -39,12 +39,31 @@
 import time
 import logging
 import logging.config
+import inspect
 
 import rospkg
 from rospkg.environment import ROS_LOG_DIR
 
 class LoggingException(Exception): pass
 
+class RospyLogger(logging.Logger):
+    def findCaller(self):
+        """
+        Find the stack frame of the caller so that we can note the source
+        file name, line number, and function name with class name if possible.
+        """
+        frame = inspect.stack()[3][0]
+        file_name = inspect.getabsfile(frame)
+        func_name = frame.f_code.co_name
+        try:
+            class_name = frame.f_locals['self'].__class__.__name__
+            func_name = '%s::%s' % (class_name, func_name)
```

Comment:
`%s::%s` is not the Python way to reference a method of a class. Please use `%s.%s` instead.